Formula/kong.rb

```diff
@@ -40,40 +55,98 @@ class Kong < Formula
        end
   PATCH
 
+  def fix_dylib_references(install_map, executable_path)
+    `otool -L #{executable_path}`.scan(/(?<=\t)(.*)(?= \(.*\n)/) do |paths|
+      old_path = paths[0]
+      lib_name = old_path.sub(%r{.*/}, "")
+      new_path = install_map[lib_name]
+      # homebrew wants us to use the macho library, but it does not work here for some reason
+      system "/usr/bin/install_name_tool", "-change", old_path, new_path, executable_path if new_path
+    end
+  end
+
   def install
-    openresty_prefix = Formula["kong/kong/openresty@#{KONG_OPENRESTY_VERSION}"].prefix
+    tmpdir = format("%<prefix>s/kong-build.%<random>f.%<now>i",
+                    prefix: ENV["HOMEBREW_TEMP"],
+                    random: rand,
+                    now:    Time.now.to_i)
+    bazel_output_user_root = "--output_user_root=#{tmpdir}/bazel"
 
-    luarocks_prefix = openresty_prefix + "luarocks"
-    openssl_prefix = openresty_prefix + "openssl"
+    # Build kong, carefully setting the environment so that brew and bazel cooperate
+    python_prefix = `brew --prefix python`.strip
+    coreutils_prefix = `brew --prefix coreutils`.strip
+    path = ENV["PATH"]
 
-    bin.install_symlink "#{openresty_prefix}/openresty/nginx/sbin/nginx"
-    bin.install_symlink "#{openresty_prefix}/openresty/bin/openresty"
-    bin.install_symlink "#{openresty_prefix}/openresty/bin/resty"
-    bin.install_symlink "#{luarocks_prefix}/bin/luarocks"
+    with_env(
+      "HOME" => "#{tmpdir}/home",
+      "PATH" => "#{python_prefix}/libexec/bin:#{coreutils_prefix}/libexec/gnubin:/usr/bin:#{path}",
+    ) do
+      system "bazel",
+             bazel_output_user_root,
+             "build",
+             "--config=release",
+             "//build:kong",
+             "--action_env=HOME",
+             "--action_env=INSTALL_DESTDIR=#{prefix}",
+             "--verbose_failures"
+    end
 
-    prefix.install "kong/include"
+    prefix.install Dir["bazel-bin/build/kong-dev/*"]
+    include.install "kong/include/opentelemetry"
+
+    bin.install "bin/kong"
+    bin.install_symlink "#{prefix}/openresty/bin/opm"
```

Comment:
slightly concerned about openresty `bin` are being default installed. Since there's openresty formula that also install those files. There're some patterns that i know:
- use `keg_only` that user needs to for `brew link kong` to install all the stuff (one more command to type for user)
- use some auto detection (https://github.com/Homebrew/homebrew-core/blob/b8fa80ab873f8943f2b0ac2cb5191b2a4d49bbf1/Formula/coreutils.rb#L830
- mark this formula as conflicts with `openresty` formula
or we can just skip installing openresty binaries at all : )

Reply:
@bungle asked for the OpenResty binaries to be installed, but I agree - It would be simpler to not install these binaries and have the user change their PATH if they need them.  Opinions?

Reply:
If we do want those binaries to be usable, I would say either mark this as a `keg_only` and tell user to do a `brew link kong` manually (like https://github.com/Homebrew/homebrew-core/blob/073c7611742d615d2cfd4d3966aee1b426808c4d/Formula/jpeg.rb#L25); or just emit an message telling user what `PATH` variable should add to be able to use those binaries. I don't see a reason they shouldn't be installed, but install them by default doesn't seem very sane.

Reply:
> I don't see a reason they shouldn't be installed, but install them by default doesn't seem very sane.

I agree.  I'll go for the message and will change the formula so that by default, only the `kong` command is in the `PATH`.  I also find it very nice that we install a `luarocks` on the standard `PATH`, but I'm not sure if that can be changed easily (e.g. by naming it `luarocks-kong` instead?)

Reply:
Yes, i think having a `luarocks-kong` in PATH makes sense, it will be useful to install dev dependencies like busted etc.

Reply:
There are probably multiple conflicting packages, especially `nginx`, `openresty`, `luarocks`.

Reply:
I've done some more investigation and found this more difficult to really get right than I had hoped.  I think for normal users who just want to try out Kong on their local machine, only the `kong` command is required.  That command uses a shebang line of `#!/usr/bin/env resty` so it expects `resty` to be on the path.  We could now change the shebang line in our homebrew version of kong so that it points into the Cellar instead. But that would still require more changes because we currently install everything that is in the build output directory under our prefix, and home links everything under `bin/` into `/opt/homebrew/bin` or equivalent.

At this point, I think I will just defer the decision how to improve the formula further until we have some feedback from users.  I will mark the formula to be conflicting with `openresty`.  If we hear back from users that this is a problem we should change the build so that the only executable that we install is `kong` and the other tools are available in an extra directory that developers need to put into their `PATH` if they want to use them.